magda-web-client/src/Components/Dataset/MetadataExtraction/extractExtents.ts

```diff
@@ -132,81 +95,93 @@ type SpatialExtent = {
 
 function aggregateDates(rows: any[], headers: string[]) {
     const dateHeaders = tryFilterHeaders(headers, DATE_REGEX);
+    const startDateHeaders = uniq(tryFilterHeaders(headers, START_DATE_REGEX));
+    const endDateHeaders = uniq(tryFilterHeaders(headers, END_DATE_REGEX));
 
-    const startDateHeaders = tryFilterHeaders(headers, START_DATE_REGEX);
-    const endDateHeaders = tryFilterHeaders(headers, END_DATE_REGEX);
     const startDateHeadersInOrder = uniq(
         startDateHeaders.concat(dateHeaders).concat(endDateHeaders)
     );
+
     const endDateHeadersInOrder = uniq(
         endDateHeaders.concat(dateHeaders).concat(startDateHeaders)
     );
 
-    console.log(
-        "Start Date Headers: " + JSON.stringify(startDateHeadersInOrder)
-    );
-    console.log("End Date Headers: " + JSON.stringify(endDateHeadersInOrder));
+    let earliestDate = maxDate;
+    let latestDate = minDate;
 
-    const dateAgg = rows.reduce(
-        (soFar: DateAggregation, row: any) => {
-            return {
-                earliestStart: startDateHeadersInOrder.reduce(
-                    (earliestStart: Date, header: string) =>
-                        pickMoment(row[header], earliestStart, (date1, date2) =>
-                            date1.getTime() <= date2.getTime() ? date1 : date2
-                        ),
-                    soFar.earliestStart
-                ),
-                latestEnd: endDateHeadersInOrder.reduce(
-                    (latestEnd: Date, header: string) =>
-                        pickMoment(row[header], latestEnd, (date1, date2) =>
-                            date1.getTime() > date2.getTime() ? date1 : date2
-                        ),
-                    soFar.latestEnd
-                )
-            };
-        },
-        {
-            earliestStart: LATEST_POSSIBLE_MOMENT,
-            latestEnd: EARLIEST_POSSIBLE_MOMENT
-        } as DateAggregation
-    );
-    const { earliestStart, latestEnd } = dateAgg;
+    startDateHeadersInOrder.forEach(header => {
+        rows.forEach(row => {
+            var dateStr: string = row[header].toString();
+            var parsedDate: Moment = undefined;
 
-    const earliestNotFound =
-        earliestStart.getTime() === LATEST_POSSIBLE_MOMENT.getTime();
-    const latestNotFound =
-        latestEnd.getTime() === EARLIEST_POSSIBLE_MOMENT.getTime();
+            // Special casing year: Explicitly specifying the year format for dates that only contain a year
+            // More context: https://github.com/moment/moment/issues/1407
+            if (header.toLowerCase() === "year") {
```

Comment:
Probably more flexible if we:
```js
let parsedDate = moment(dateStr, "YYYY", true);
if(!parsedDate.isValid()){
  parsedDate = moment(dateStr);
}
```
So that we don't need to check the column name